From 2d91f9fbbec5420ded205cd9fd363735ae16e2ce Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Fri, 22 Apr 2022 20:18:14 +0200
Subject: [PATCH 41/94] arm64: dts: imx8mp: Enable HDMI on MX8MP DHCOM PDK2

Enable HDMI output on MX8MP DHCOM PDK2. The I2C5 is used in regular I2C
mode instead of HDMI DDC mode to permit connection of other I2C devices
on that bus.

Upstream-Status: Pending [Depends on MX8MP HDMI support]
Signed-off-by: Marek Vasut <marex@denx.de>
Cc: Fabio Estevam <festevam@denx.de>
Cc: Peng Fan <peng.fan@nxp.com>
Cc: Shawn Guo <shawnguo@kernel.org>
Cc: NXP Linux Team <linux-imx@nxp.com>
To: linux-arm-kernel@lists.infradead.org
---
 arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts
index fdf18b7f208d5..d4c1eebb35015 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts
@@ -151,6 +151,17 @@ &flexcan1 {
 	status = "okay";
 };
 
+&hdmi_tx {
+	ddc-i2c-bus = <&i2c5>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi>;
+	status = "okay";
+};
+
+&hdmi_tx_phy {
+	status = "okay";
+};
+
 &pcie_phy {
 	clock-names = "ref";
 	clocks = <&clk IMX8MP_SYS_PLL2_100M>;
-- 
2.39.2

